include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -372,108 +375,124 @@ namespace LinearAlgebra
+                exec, 0, n_elements),
+              KOKKOS_LAMBDA(size_type i) {
+                data.values(indices_dev(i)) += tmp_vector(i);
+              });
           else
-            ::dealii::LinearAlgebra::CUDAWrappers::kernel::set_permutated<
-              Number><<<n_blocks, ::dealii::CUDAWrappers::block_size>>>(
-              indices_dev,
-              data.values.data(),
-              tmp_vector.begin(),
-              tmp_n_elements);
-
-          ::dealii::Utilities::CUDA::free(indices_dev);
-          ::dealii::Utilities::CUDA::free(V_dev);
+            Kokkos::parallel_for(
+              "import_elements: set values",
+              Kokkos::RangePolicy<
+                ::dealii::MemorySpace::Default::kokkos_space::execution_space>(
+                exec, 0, n_elements),
+              KOKKOS_LAMBDA(size_type i) {
+                data.values(indices_dev(i)) = tmp_vector(i);
+              });
+          exec.fence();
         }
 
         template <typename RealType>
         static void
         linfty_norm_local(const ::dealii::MemorySpace::MemorySpaceData<
                             Number,
-                            ::dealii::MemorySpace::CUDA> &data,
-                          const unsigned int              size,
-                          RealType &                      result)
+                            ::dealii::MemorySpace::Default> &data,
+                          const unsigned int                 size,
+                          RealType &                         result)
         {
           static_assert(std::is_same<Number, RealType>::value,
                         "RealType should be the same type as Number");
 
-          Number *    result_device;
-          cudaError_t error_code = cudaMalloc(&result_device, sizeof(Number));
-          AssertCuda(error_code);
-          error_code = cudaMemset(result_device, 0, sizeof(Number));
-
-          const int n_blocks = 1 + size / (::dealii::CUDAWrappers::chunk_size *
-                                           ::dealii::CUDAWrappers::block_size);
-          ::dealii::LinearAlgebra::CUDAWrappers::kernel::reduction<
-            Number,
-            ::dealii::LinearAlgebra::CUDAWrappers::kernel::LInfty<Number>>
-            <<<dim3(n_blocks, 1), dim3(::dealii::CUDAWrappers::block_size)>>>(
-              result_device, data.values.data(), size);
-
-          // Copy the result back to the host
-          error_code = cudaMemcpy(&result,
-                                  result_device,
-                                  sizeof(Number),
-                                  cudaMemcpyDeviceToHost);
-          AssertCuda(error_code);
-          // Free the memory on the device
-          error_code = cudaFree(result_device);
-          AssertCuda(error_code);
+          typename ::dealii::MemorySpace::Default::kokkos_space::execution_space
+            exec;
+          Kokkos::parallel_reduce(
```

Comment:
You forgot to use a name for all the `parallel_reduce`

Reply:
Fixed.